preferences/views.py

```diff
@@ -0,0 +1,43 @@
+from django.shortcuts import render, redirect, reverse
+from django.contrib.auth.decorators import login_required
+from preferences.forms import PreferencesForm
+from preferences.models import Preferences
+from django.core import serializers
+
+# this function creates a blank preference model form on GET request
+# and returns a context with the form and the fields from the preference
+# object tagged to the current user
+# on non GET request we update the database to save the new preferences
+@login_required
+def setPreferences (request):
+    context = {}
+    if request.method == 'GET':
+        context['form'] = PreferencesForm ()
+        # null check for preferences field
+        if request.user.preferences:
+            fields = [(field.name, field.value_to_string(request.user.preferences)) for field in Preferences._meta.fields]
+            context['fields'] = fields
```

Comment:
I think we only need to return a form instance pre-populated with the existing user preferences.Django takes care of everything else.
```suggestion
            context['form'] = PreferencesForm(instance=request.user.preferences)
```